docs/ref/models/expressions.txt

```diff
@@ -615,6 +613,12 @@ This is the only way to perform an aggregation within a ``Subquery``, as
 using :meth:`~.QuerySet.aggregate` attempts to evaluate the queryset (and if
 there is an ``OuterRef``, this will not be possible to resolve).
 
+Using Expressions to filter QuerySets
+-------------------------------------
+
+When an Expression outputs a boolean value, it may be used to directly
+filter a queryset, or as the argument to a ``When`` expression.
```

Comment:
In fact, it's not all conditional queries, but only those that return a boolean value.